### DISCUSSION TITLE
Help requested: writing a converter to add device functionality to z2m with information from sniffed Zigbee traffic

### DISCUSSION THREAD

**Discussion by Tediore** [Koenkk/zigbee2mqtt#8509] [Devices]
It's possible to customize the behavior of the built-in siren and LED on the Linkind water leak sensor (https://github.com/Koenkk/zigbee-herdsman-converters/commit/e671c0a331a76f739fe793e8abb9df8c8a4d7b8e) with the official hub. I sniffed the traffic between the Linkind hub and the sensor when selecting the four possible settings in the Linkind app, but I'm not sure how to write a converter in `toZigbee.js` for it. The hub writes an int to endpoint 1, basic cluster, attr 0x400a on the sensor. Here are more details:
```
Endpoint                    1
Cluster                     Basic (0x0000)
Attribute                   0x400a
ACK request                 TRUE
Manuf. specific             TRUE
Disable default response    TRUE
Data type                   8-bit unsigned int

Data:    
  Siren disabled, LED disabled     0 (0x00)
  Siren disabled, LED enabled      1 (0x01)
  Siren enabled, LED disabled      2 (0x02)
  Siren enabled, LED enabled       3 (0x03)
```

I took a look at `toZigbee.js` and I don't even know where to start with adding a converter. Thankfully the default behavior is siren and LED enabled, and I don't wish to change this behavior; I just thought it would be fun to sniff the Zigbee messages and add support for these options to z2m. Would somebody be able to help me do this?

**Comment by Koenkk** [comment]
https://github.com/Koenkk/zigbee-herdsman-converters/pull/2953/files should do it.

- please replace todo with the manufacturerCode (should also be in the sniff) (`manufacturerCode: TODO`)
- please check the terminology, I called it `alarm_behaviour`, not sure if that is a good name.

**Reply by Tediore** [reply]
Awesome, thanks! Can confirm it's working (after adding the missing semicolon). I have the manufacturer code (`0x1168`), and I would also suggest changing `alarm_behaviour` to `alert_behaviour`. I can't make edits to your PR though; would you like me to make those changes in my fork and submit it as a PR? The sensor also supports battery percentage, so I was going to add that functionality too. Might as well just roll it into the same PR.

**Reply by Tediore** [reply]
PR created :smile:

Figured the answer was likely obvious.
